### PR TITLE
Refactored Keeper Integration Tests to Use Table tests

### PR DESCRIPTION
**Pull request by vla33**

### CODE REVIEW

File: integration-tests/smoke/keeper_test.go [hunk 1]
```diff
@@ -115,549 +115,538 @@ func getKeeperSuite(
...
+					"env": map[string]interface{}{
+						"MIN_INCOMING_CONFIRMATIONS": "1",
+						"KEEPER_TURN_FLAG_ENABLED":   "true",
+						"KEEPER_TURN_LOOK_BACK":      "0",
+					},
+				}))
+			err = testEnvironment.Run()
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("Connecting to launched resources")
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:140]: not sure i understand this. Why this change? (removing func() from By)

File: integration-tests/smoke/keeper_test.go [hunk 2]
```diff
@@ -115,549 +115,538 @@ func getKeeperSuite(
...
-				Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
-			})
+			By("Register Keeper Jobs")
+			actions.CreateKeeperJobs(chainlinkNodes, registry)
+			err = chainClient.WaitForEvents()
+			Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
 		})
 
-		Describe("with Keeper job", func() {
+		DescribeTable("with Keeper job", func() {
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:194]: I'm not sure if DescribeTable has any difference than Describe in its current form. Looking at [example](https://pkg.go.dev/github.com/onsi/ginkgo/extensions/table) it seems to be useful when the function within Describe has some parameters, and you can specify these parameters as different table entries

**Review by infiloop2 — CHANGES_REQUESTED** [review]
added inline comments

File: integration-tests/smoke/keeper_test.go (hunk 1 above)
Comment (vla33) [reply integration-tests/smoke/keeper_test.go:140]: According to the documentation [here](https://onsi.github.io/ginkgo/#documenting-complex-specs-by), it seems that `func()` is an optional argument. The other test suites (VRF, OCR etc.) don't seem to use it anymore either after the refactoring.

File: integration-tests/smoke/keeper_test.go [hunk 3]
```diff
@@ -95,569 +66,797 @@ var highBCPTRegistryConfig = contracts.KeeperRegistrySettings{
...
-						contractDeployer,
-						chainClient,
-						&registryConfig,
-						linkFundsForEachUpkeep,
-						10000,   // How many blocks this upkeep will be eligible from first upkeep block
-						5,       // Interval of blocks that upkeeps are expected to be performed
-						100000,  // How much gas should be burned on checkUpkeep() calls
-						4000000, // How much gas should be burned on performUpkeep() calls. Initially set higher than defaultUpkeepGasLimit
-					)
+var _ = Describe("Keeper Suite @keeper", func() {
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:69]: should this be describe table?

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Not sure why we need a describe table within describe. Can we have a top level describeTable?

File: integration-tests/smoke/keeper_test.go [hunk 4]
```diff
@@ -95,569 +66,797 @@ var highBCPTRegistryConfig = contracts.KeeperRegistrySettings{
...
+				100000,  // How much gas should be burned on checkUpkeep() calls
+				4000000, // How much gas should be burned on performUpkeep() calls. Initially set higher than defaultUpkeepGasLimit
+			)
+		}
+
+		By("Register Keeper Jobs")
+		actions.CreateKeeperJobs(chainlinkNodes, registry)
+		err = chainClient.WaitForEvents()
+		Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
+
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:848]: why different from beforeEach?

File: integration-tests/smoke/keeper_test.go [hunk 5]
```diff
@@ -95,569 +80,577 @@ var highBCPTRegistryConfig = contracts.KeeperRegistrySettings{
...
+			Entry("v1.1 Add funds to upkeep test @simulated", ethereum.RegistryVersion_1_1, defaultRegistryConfig, BasicCounter, AddFundsToUpkeepTest, big.NewInt(1)),
+			Entry("v1.2 Add funds to upkeep test @simulated", ethereum.RegistryVersion_1_2, defaultRegistryConfig, BasicCounter, AddFundsToUpkeepTest, big.NewInt(1)),
+			Entry("v1.1 Removing one keeper test @simulated", ethereum.RegistryVersion_1_1, defaultRegistryConfig, BasicCounter, RemovingKeeperTest, big.NewInt(defaultLinkFunds)),
+			Entry("v1.2 Removing one keeper test @simulated", ethereum.RegistryVersion_1_2, defaultRegistryConfig, BasicCounter, RemovingKeeperTest, big.NewInt(defaultLinkFunds)),
+			Entry("v1.2 Pause registry test @simulated", ethereum.RegistryVersion_1_2, defaultRegistryConfig, BasicCounter, PauseRegistryTest, big.NewInt(defaultLinkFunds)),
+			Entry("v1.2 Migrate upkeep from a registry to another @simulated", ethereum.RegistryVersion_1_2, defaultRegistryConfig, BasicCounter, MigrateUpkeepTest, big.NewInt(defaultLinkFunds)),
+		}
+
+		runSpecificTest = func(registryVersion ethereum.KeeperRegistryVersion, registryConfig contracts.KeeperRegistrySettings,
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:114]: You can probably move this directly within the DescribeTable?

**Review by infiloop2 — DISMISSED** [review]
lgtm! Thanks for diving deep into this

File: integration-tests/smoke/keeper_test.go (hunk 3 above)
Comment (vla33) [reply integration-tests/smoke/keeper_test.go:83]: Yes, we could instead have a DescribeTable here and then make the variables global, but in order to stay consistent with the other test suites (like VRF and OCR) we should use this format.

**Review by infiloop2 — APPROVED** [review]
